packages/lab/src/banner/Banner.tsx

```diff
@@ -1,100 +1,49 @@
+import { forwardRef, HTMLAttributes, useEffect, useState } from "react";
 import {
-  ComponentType,
-  forwardRef,
-  HTMLAttributes,
-  MouseEvent,
-  ReactNode,
-  useEffect,
-  useState,
-  Ref,
-} from "react";
-import {
-  Button,
-  ButtonProps,
-  Link,
-  LinkProps,
   makePrefixer,
   StatusIndicator,
   useAriaAnnouncer,
   useForkRef,
   ValidationStatus,
 } from "@salt-ds/core";
-
-import getInnerText from "./internal/getInnerText";
-import { CloseIcon, IconProps } from "@salt-ds/icons";
 import { clsx } from "clsx";
 
+import getInnerText from "./internal/getInnerText";
 import "./Banner.css";
 
-export type BannerStatus = ValidationStatus;
-
-export type LabelProps = { className?: string };
-
-type StateAndPropsGetterFunction<TInjectedProps> = <T>(
-  props?: T
-) => TInjectedProps;
-
-export interface GetStateAndPropGetters {
-  Icon: ComponentType<IconProps>;
-  getIconProps: StateAndPropsGetterFunction<IconProps>;
-  getLabelProps: StateAndPropsGetterFunction<LabelProps>;
-  getLinkProps: StateAndPropsGetterFunction<LinkProps>;
-}
-
 export interface BannerProps extends HTMLAttributes<HTMLDivElement> {
-  /**
-   * THe props to be passed to the close button
-   */
-  CloseButtonProps?: ButtonProps;
-  /**
-   * Props to pass to Link, Link will not be rendered if this is undefined
-   */
-  LinkProps?: LinkProps;
   /**
    * Announcement message for screen reader 250ms after the banner is mounted.
    */
   announcement?: string;
   /**
-   * close button ref
-   */
-  closeRef?: Ref<HTMLButtonElement>;
-  /**
-   * If true, the built-in ARIA announcer will be disabled
+   * If false, the built-in ARIA announcer will be applied. If true, the default browser behaviour will be applied instead.
    */
   disableAnnouncer?: boolean;
   /**
    * Emphasize the styling by applying a background color: defaults to false
    */
   emphasize?: boolean;
   /**
-   * onClose callback when the close icon is clicked, the dismiss button will not be rendered if this is
-   * not defined
-   */
-  onClose?: (event: MouseEvent<HTMLButtonElement>) => void;
-  /**
-   * render props callback (if children callback not supplied)
+   * Controlled prop to set visible state.
    */
-  render?: (props: GetStateAndPropGetters) => ReactNode;
+  open?: boolean;
```

Comment:
Just a question but is it still worth having an open prop versus conditionally mounting the component?

Reply:
Most probably we don't :)